src/avatar/src/Avatar.js

```diff
@@ -21,167 +21,145 @@ const initialsStyleClass = css({
   lineHeight: 1
 }).toString()
 
-class Avatar extends PureComponent {
-  static propTypes = {
-    /**
-     * The src attribute of the image.
-     * When it's not available, render initials instead.
-     */
-    src: PropTypes.string,
-
-    /**
-     * The size of the avatar.
-     */
-    size: PropTypes.number,
-
-    /**
-     * The name used for the initials and title attribute.
-     */
-    name: PropTypes.string,
-
-    /**
-     * The value used for the hash function.
-     * The name is used as the hashValue by default.
-     * When dealing with anonymous users you should use the id instead.
-     */
-    hashValue: PropTypes.string,
-
-    /**
-     * When true, render a solid avatar.
-     */
-    isSolid: PropTypes.bool,
-
-    /**
-     * The color used for the avatar.
-     * When the value is `automatic`, use the hash function to determine the color.
-     */
-    color: PropTypes.string.isRequired,
-
-    /**
-     * Function to get the initials based on the name.
-     */
-    getInitials: PropTypes.func,
-
-    /**
-     * When true, force show the initials.
-     * This is useful in some cases when using Gravatar and transparent pngs.
-     */
-    forceShowInitials: PropTypes.bool,
-
-    /**
-     * When the size is smaller than this number, use a single initial for the avatar.
-     */
-    sizeLimitOneCharacter: PropTypes.number,
-
-    /**
-     * Theme provided by ThemeProvider.
-     */
-    theme: PropTypes.object.isRequired
+function getColorProps({ isSolid, theme, color, name, propsHashValue }) {
+  if (color === 'automatic') {
+    const hashValue = globalHash(propsHashValue || name)
+    return theme.getAvatarProps({ isSolid, color, hashValue })
   }
 
-  static defaultProps = {
-    color: 'automatic',
-    size: 24,
-    isSolid: false,
-    getInitials: globalGetInitials,
-    forceShowInitials: false,
-    sizeLimitOneCharacter: 20
-  }
+  return theme.getAvatarProps({ isSolid, color })
+}
 
-  constructor(props, context) {
-    super(props, context)
-    this.state = { imageHasFailedLoading: false }
+function Avatar({
```

Comment:
For `defaultProps` should those just be assigned in the method signature? e.g. `forceShowInitials = false`